src/pages/SignUpPage/SignupPage.jsx

```diff
@@ -102,29 +103,37 @@ function SignupPage() {
                 "firstName": state.firstName,
                 "lastName": state.lastName
             };
-            console.log(object)
-            var res = await axios.post("/api/auth/signup", object);
-            console.log(res);
-            if (res.status === 201) {
-                var token = res?.data?.encodedToken;
-                localStorage.setItem(VAR_ENCODE_TOKEN, token)
-                var user = res?.data?.createdUser;
-                var userId = res?.data?.createdUser?._id;
-                localStorage.setItem(VAR_USER_ID, userId);
-                let userDetails = { email: res.data.createdUser.email, firstName: res.data.createdUser.firstName, lastName: res.data.createdUser.lastName };
-                localStorage.setItem(VAR_USER_DETAILS, JSON.stringify(userDetails));
-                userDispatch(userDetails)
-                console.log(user, userId, token);
-                setlogin(true);
-                
-                // History.push("/products");
+            console.log(VerifyCredentials())
+           
+                console.log(object)
+                var res = await axios.post("/api/auth/signup", object);
+                console.log(res);
+                if (res.status === 201) {
+                    var token = res?.data?.encodedToken;
+                    localStorage.setItem(VAR_ENCODE_TOKEN, token)
+                    var user = res?.data?.createdUser;
+                    var userId = res?.data?.createdUser?._id;
```

Comment:
I am not sure but do you need actually three-level optional chaining.

Reply:
If there is issue within backend and it doesn't give data or created User so here it shouldn't fail and handle